types/extrinsic.go

```diff
@@ -51,11 +68,14 @@ type Extrinsic struct {
 	Method Call
 }
 
-// NewExtrinsic creates a new Extrinsic from the provided Call
-func NewExtrinsic(c Call) Extrinsic {
-	return Extrinsic{
-		Version: ExtrinsicVersion4,
-		Method:  c,
+// NewExtrinsic creates a new Extrinsic from the provided CallWrapper
+func NewExtrinsic(c CallWrapper) ExtrinsicWrapper {
```

Comment:
I am thinking on renaming this to (or reusing above `NewExtrinsicWrapper`) So it is more clear what it returns.